kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/RecordEncryptionST.java

```diff
@@ -115,24 +128,126 @@ void produceAndConsumeMessage(String namespace, TestKmsFacade<?, ?, ?> testKmsFa
         int numberOfMessages = 1;
 
         // start Kroxylicious
-        LOGGER.atInfo().setMessage("Given Kroxylicious in {} namespace with {} replicas").addArgument(namespace).addArgument(1).log();
+        LOGGER.info("Given Kroxylicious in {} namespace with {} replicas", namespace, 1);
         Kroxylicious kroxylicious = new Kroxylicious(namespace);
         kroxylicious.deployPortPerBrokerPlainWithRecordEncryptionFilter(clusterName, 1, testKmsFacade);
         bootstrap = kroxylicious.getBootstrap();
 
-        LOGGER.atInfo().setMessage("And a kafka Topic named {}").addArgument(topicName).log();
+        LOGGER.info("And a kafka Topic named {}", topicName);
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 2);
+
+        LOGGER.info("When {} messages '{}' are sent to the topic '{}'", numberOfMessages, MESSAGE, topicName);
+        KroxyliciousSteps.produceMessages(namespace, topicName, bootstrap, MESSAGE, numberOfMessages);
+
+        LOGGER.info("Then the messages are consumed");
+        List<ConsumerRecord> result = KroxyliciousSteps.consumeMessages(namespace, topicName, bootstrap, numberOfMessages, Duration.ofMinutes(2));
+        LOGGER.info("Received: {}", result);
+
+        assertThat(result).withFailMessage("expected messages have not been received!")
+                .extracting(ConsumerRecord::getValue)
+                .hasSize(numberOfMessages)
+                .allSatisfy(v -> assertThat(v).contains(MESSAGE));
+    }
+
+    @TestTemplate
+    void ensureClusterHasEncryptedMessageWithRotatedKEK(String namespace, TestKmsFacade<?, ?, ?> testKmsFacade) {
+        // Skip AWS test execution because the ciphertext blob metadata to read the version of the KEK is not available anywhere
+        assumeThat(testKmsFacade.getKmsServiceClass().getSimpleName().toLowerCase().contains("vault")).isTrue();
+        testKekManager = testKmsFacade.getTestKekManager();
+        testKekManager.generateKek("KEK_" + topicName);
+        int numberOfMessages = 1;
+        ExperimentalKmsConfig experimentalKmsConfig = new ExperimentalKmsConfig(1, 1, 1, 1);
```

Comment:
Passing `(1, 1, 1, 1)` makes it really unclear what the test is actually testing.

I'd suggest setting encryptionDekExpireAfterWriteSeconds to value like `5s` and set the others to `null` so you leave their default values in place.

After the first consume, you'll need a `Thread.sleep(5000)` to make sure that Caffeine has had chance to expire the entry.

I needed to read https://github.com/ben-manes/caffeine/wiki/Refresh to remind myself how refresh and expire actually work.

There is be an argument for system testing both refresh and expire.  However I think just one is sufficient.

Reply:
I set all values because at the beginning it was a try to run the test for all KMS, so we need all values to be set. But yes, actually now it is only needed the one related to Vault.